src/ukrdc_cupid/core/store/insert.py

```diff
@@ -0,0 +1,123 @@
+import time
+
+from sqlalchemy.orm import Session
+from ukrdc_cupid.core.store.models.ukrdc import PatientRecord
+from sqlalchemy.exc import OperationalError
+from ukrdc_sqla.ukrdc import Base
+
+import ukrdc_xsdata.ukrdc as xsd_ukrdc  # type: ignore
+from typing import Optional, Tuple
+
+
+def advisory_lock(func):
+    def wrapper(ukrdc_session: Session, pid: str, *args, **kwargs):
+        # this wrapper function applies for an advisory lock on a a particular pid
+        # if it isn't available it will keep trying for up to 60 secs to obtain the lock
+
+        start_time = time.time()
+        max_wait_time = 60  # Maximum wait time in seconds
+
+        while time.time() - start_time < max_wait_time:
+            try:
+                # Try to acquire an advisory lock for the specified pid
+                ukrdc_session.execute(
+                    f"SELECT pg_advisory_lock({int(pid)}::int)"
```

Comment:
Not sure the Python int() is neccesary?

Reply:
Think its code distortion caused by static tester totalitarianism.